File: micro_bit/skrecanie-w-miejscu-3/main.py
AlphaBot2.run(Dir.STOP, 0)
basic.show_arrow(ArrowNames.WEST)

def on_forever():
    basic.show_icon(IconNames.YES)
    while True:
        if AlphaBot2.ultrasonic() > 10:
            AlphaBot2.run(Dir.FORWARD, 40)
        else:
            AlphaBot2.motor_run(Motors.M1, -40)
            basic.pause(500)
basic.forever(on_forever)

File: micro_bit/skrecanie-w-miejscu-3/main.ts
input.onButtonPressed(Button.A, function () {
    basic.showIcon(IconNames.Yes)
    while (true) {
        if (AlphaBot2.Ultrasonic() > 10) {
            AlphaBot2.Run(Dir.forward, 40)
        } else {
            AlphaBot2.MotorRun(Motors.M1, -40)
            basic.pause(500)
        }
    }
})
AlphaBot2.Run(Dir.stop, 0)
basic.showArrow(ArrowNames.West)
